visivo/models/model.py

```diff
@@ -1,12 +1,53 @@
-from typing import Optional
-from pydantic import Field, validator
+from typing import Any, List, Optional, Union
+from typing_extensions import Annotated
+from pydantic import Field, Discriminator, Tag
 from visivo.models.base.named_model import NamedModel
 from visivo.models.base.parent_model import ParentModel
-from visivo.models.target import DefaultTarget, Target
-from .base.base_model import generate_ref_field
+from visivo.models.target import DefaultTarget, Target, TypeEnum
+from .base.base_model import RefString, generate_ref_field
+import pandas
+import re
 
 
-class Model(NamedModel, ParentModel):
+class Model(NamedModel):
+    pass
+
+
+class RunModel(Model):
+    cmds: List[str] = Field(description="The sql used to generate your base data")
+
+    @property
+    def sql(self):
+        return f"select * from {self.table_name}"
+
+    @property
+    def table_name(self):
+        return re.sub(r"[^a-zA-Z0-9_]", "_", self.name)
```

Comment:
I wonder if we should throw a warning if they use non standard characters to let them know the table name is different than what they might expect.

Reply:
I added a last commit to constrain the name to be valid, is that better?